torch_xla/csrc/tensor.cpp

```diff
@@ -904,8 +904,11 @@ XLATensor XLATensor::full_like(const XLATensor& input, at::Scalar fill_value,
 
 XLATensor XLATensor::select(const XLATensor& input, int64_t dim,
                             int64_t index) {
-  return Create(ir::MakeNode<ir::ops::Select>(input.GetIrValue(), dim, index),
-                input.GetDevice());
+  auto input_shape = input.shape();
+  int select_dim = GetCanonicalDimensionIndex(dim, input_shape.get().rank());
```

Comment:
I'd move this outta tensor.cpp into helpers.cpp

Reply:
The `GetCanonicalDimensionIndex` helper or something else?

Reply:
Yep, that one.

Reply:
Done.